### PR TITLE
[lightbeam] Updates dependencies

### PR DESCRIPTION
**Pull request by 0x7CFE**
Updates dependencies of Lightbeam.

Note that [the `multi_mut` project](https://github.com/golddranks/multi_mut) was not updated for like 3 years (latest commit from 2017). Looks like the author had abandoned the project and is not replying to issues. However, it's unpublished `master` branch contains [fixes that theoretically should improve soundness](https://github.com/golddranks/multi_mut/commit/561e9f7506e8aeddd7ee672d0f844437babe3da8). Instead of direct transmutes of references it uses raw pointers. Whether it's at all sound remains an open question. For details [see the disclaimer](https://github.com/golddranks/multi_mut#disclaimer) of the project itself.

The whole `multi_mut` approach seems very tricky and error prone to me. Currently I'm investigating how corresponding Lightbeam logic could be rewritten so that it would no longer need `multi_mut`. It's [used in a single place](https://github.com/bytecodealliance/wasmtime/blob/master/crates/lightbeam/src/function_body.rs#L373) so it's probably doable.

### CODE REVIEW

**Comment by Vurich** [comment]
If you look at the commit on my personal fork https://github.com/Vurich/wasmtime/commit/545a39836fb812783bb0da2b087dfb42f6e451dc, this is the way that I suggest removing `multi_mut`. Unfortunately as the migration to the current Wasmtime master branch is incomplete it's not possible to fully test this change, but it doesn't result in any additional tests failing or failing with different errors.

**Comment by 0x7CFE** [comment]
> this is the way that I suggest removing `multi_mut`. 

Yes, this approach is much more robust. Honestly, I don't think that it would really affect performance that much.

> Unfortunately as the migration to the current Wasmtime master branch is incomplete it's not possible to fully test this change

What is the reason for the stall? May I take the migration task?

> …but it doesn't result in any additional tests failing or failing with different errors.

In that case, since it's an independent change, I would suggest filing it as a separate PR and merge it early.

File: crates/lightbeam/Cargo.toml [hunk 1]
```diff
@@ -11,24 +11,24 @@ keywords = ["webassembly", "wasm", "compile", "compiler", "jit"]
...
+dynasmrt = "0.6.0"
 wasmparser = "0.51.2"
 memoffset = "0.5.3"
 itertools = "0.8.2"
 capstone = "0.6.0"
-thiserror = "1.0.9"
+thiserror = "1.0.11"
 cranelift-codegen = "0.59.0"
-multi_mut = "0.1"
+multi_mut = { git = "https://github.com/golddranks/multi_mut", rev = "561e9f7506e8aeddd7ee672d0f844437babe3da8" }
```
Comment (bjorn3) [inline crates/lightbeam/Cargo.toml]: This would prevent this crate from being published on crates.io.

Comment (0x7CFE) [reply crates/lightbeam/Cargo.toml]: Thanks for commenting, @bjorn3. Yes, I've already figured this out. Looks like @Vurich had proposed better approach that would eliminate `multi_mut` completely.

Comment (sunfishcode) [reply crates/lightbeam/Cargo.toml]: Cool, just to confirm, we do publish on crates.io, so we'll need the better approach here :slightly_smiling_face: .

**Comment by 0x7CFE** [comment]
Closed in favor of https://github.com/bytecodealliance/wasmtime/pull/1232

**Pull request closed without merging**